I see these failures consistently on F34:

FAILED basic_test.py::test_retrocl_exclude_attr_add - assert False
FAILED basic_test.py::test_retrocl_exclude_attr_mod - ldap.SERVER_DOWN: {'result': -1, 'desc': "Can't contact LDAP server", 'ctrls': []}
FAILED basic_test.py::test_retrocl_trimming - ldap.SERVER_DOWN: {'result': -1, 'desc': "Can't contact LDAP server", 'ctrls': []}
FAILED retrocl_indexing_test.py::test_indexing_is_online - ldap.OBJECT_CLASS_VIOLATION: {'msgtype': 103, 'msgid': 6, 'result': 65, 'desc': 'Object class violation', 'ct


This stack appears related to this work:

                #0  0x00007f4fe6c89f74 entry2reple (libretrocl-plugin.so + 0x5f74)
                #1  0x00007f4fe6c8bed6 retrocl_postob (libretrocl-plugin.so + 0x7ed6)
                #2  0x00007f4febf83df9 plugin_call_func (libslapd.so.0 + 0x138df9)
                #3  0x00007f4febf84006 plugin_call_plugins (libslapd.so.0 + 0x139006)
                #4  0x00007f4fe6e7e62f ldbm_back_add (libback-ldbm.so + 0x5f62f)
                #5  0x00007f4febf2b707 op_shared_add (libslapd.so.0 + 0xe0707)
                #6  0x00007f4febf2c593 do_add (libslapd.so.0 + 0xe1593)
                #7  0x0000555f97c6ec33 connection_threadmain.lto_priv.0 (ns-slapd + 0x1ac33)
                #8  0x00007f4feba2a524 _pt_root (libnspr4.so + 0x2c524)
                #9  0x00007f4feb9bd299 start_thread (libpthread.so.0 + 0x9299)
                #10 0x00007f4febd5c353 __clone (libc.so.6 + 0x100353)


But there are a ton of other crashes, and I suspect heap corruption.  I suggest running this under an ASAN build just to make sure.

--- a/ldap/servers/plugins/retrocl/retrocl.c
+++ b/ldap/servers/plugins/retrocl/retrocl.c
@@ -731,6 +731,6 @@ retrocl_attr_in_exclude_attrs(char *attr, int attrlen)
                 i++;
             }
         }
-        return 0;
     }
+    return 0;
 }

